framework/src/util/CFRLogger.cpp

```diff
@@ -112,10 +115,38 @@ void CFRLogger::Log(const cppmicroservices::ServiceReferenceBase& sr,
                     const std::string& message,
                     const std::exception_ptr ex)
 {
-  auto currLogger = std::atomic_load(&logService);
-  if (currLogger) {
-    currLogger->Log(sr, level, message, ex);
+   if (IsOpen()) {
+    auto currLogger = std::atomic_load(&logService);
+    if (currLogger) {
+      currLogger->Log(sr, level, message, ex);
+    }
   }
 }
+
+void CFRLogger::Open()
+{
+  auto l = this->Lock();
```

Comment:
There is only ever 1 instance of CFRLogger in the process. It is opened in the constructor of `CoreBundleContext` and closed in the destructor. Additionally, the only class who uses the CFRLogger is `CoreBundleContext`. Code outside of that class should not be able to call open on it.

Why is a lock here necessary given that there is only 1 call to open and 1 call to close in the entire library/process?